web/crux/src/app/deploy/deploy.service.ts

```diff
@@ -465,39 +465,48 @@ export default class DeployService {
         id: deployment.id,
         releaseNotes: deployment.version.changelog,
         versionName: deployment.version.name,
-        requests: deployment.instances.map(it => {
-          const { registry } = it.image
-          const registryUrl =
-            registry.type === 'google' || registry.type === 'github'
-              ? `${registry.url}/${registry.imageNamePrefix}`
-              : registry.type === 'v2' || registry.type === 'gitlab' || registry.type === 'unchecked'
-              ? registry.url
-              : registry.type === 'hub'
-              ? registry.imageNamePrefix
-              : ''
-
-          const mergedConfig = mergedConfigs.get(it.id)
-
-          return {
-            common: this.mapper.commonConfigToAgentProto(mergedConfig),
-            crane: this.mapper.craneConfigToAgentProto(mergedConfig),
-            dagent: this.mapper.dagentConfigToAgentProto(mergedConfig),
-            id: it.id,
-            containerName: it.image.config.name,
-            imageName: it.image.name,
-            tag: it.image.tag,
-            instanceConfig: this.mapper.deploymentToAgentInstanceConfig(deployment),
-            registry: registryUrl,
-            registryAuth: !registry.token
-              ? undefined
-              : {
-                  name: registry.name,
-                  url: registryUrl,
-                  user: registry.user,
-                  password: registry.token,
-                },
-          } as DeployRequest
-        }),
+        requests: await Promise.all(
+          deployment.instances.map(async it => {
+            const { registry } = it.image
+            const registryUrl =
+              registry.type === 'google' || registry.type === 'github'
+                ? `${registry.url}/${registry.imageNamePrefix}`
+                : registry.type === 'v2' || registry.type === 'gitlab' || registry.type === 'unchecked'
+                ? registry.url
+                : registry.type === 'hub'
+                ? registry.imageNamePrefix
+                : ''
+
+            const mergedConfig = mergedConfigs.get(it.id)
+            const storage = mergedConfig.storageId
+              ? await this.prisma.storage.findFirstOrThrow({
```

Comment:
This should be in the ValidationPipe, i belive.

Reply:
That would require merging all the configs just to check the storage. Wouldn't that be too much overhead?